File: backend/src/main/java/com/a603/ofcourse/global/domain/Points.java
package com.a603.ofcourse.global.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Embeddable
@NoArgsConstructor
@AllArgsConstructor
public class Points {

    public static final double EARTH_RADIUS = 6371e3; // in meters

    @Column(name = "latitude")
    private double lat;   // 위도
    @Column(name = "longitude")
    private double lng;   // 경도

    private Points calculatePoints(double distance, boolean isMax) {
        double deltaLat = Math.toDegrees(distance / EARTH_RADIUS);
        double deltaLon = Math.toDegrees(distance / (EARTH_RADIUS * Math.cos(Math.toRadians(lat))));
        double newLat = lat + (isMax ? deltaLat : -deltaLat);
        double newLon = lng + (isMax ? deltaLon : -deltaLon);
        return new Points(newLat, newLon);
    }

    public Points getMinPoints(double distance) {
        return calculatePoints(distance, false);
    }

    public Points getMaxPoints(double distance) {
        return calculatePoints(distance, true);
    }

}

File: bigdata/domain/points.py
import math

EARTH_RADIUS = 6371000.0


class Points:
    def __init__(self, lat, lng):
        self.lat = lat
        self.lng = lng

    def calculate_points(self, dist, isMax):
        delta_lat = math.degrees(dist / EARTH_RADIUS)
        delta_lon = math.degrees(dist / (EARTH_RADIUS * math.cos(math.radians(self.lat))))
        new_lat = self.lat + (delta_lat if isMax else -delta_lat)
        new_lon = self.lng + (delta_lon if isMax else -delta_lon)
        return Points(new_lat, new_lon)

    def get_min_points(self, dist):
        return self.calculate_points(dist, False)

    def get_max_points(self, dist):
        return self.calculate_points(dist, True)

